log-base/src/Log/Monad.hs

```diff
@@ -42,6 +45,24 @@ newtype LogT m a = LogT { unLogT :: InnerLogT m a }
   deriving (Alternative, Applicative, Functor, Monad, MonadBase b, MonadCatch
            ,MonadIO, MonadMask, MonadPlus, MonadThrow, MonadTrans)
 
+instance MonadError e m => MonadError e (LogT m) where
+    throwError = lift . throwError
+    catchError m h = LogT $ unLogT m `catchError` (unLogT . h)
+
+instance MonadReader r m => MonadReader r (LogT m) where
```

Comment:
I think it's better to use the default `ReaderT` instance for `MonadReader` instead of this one as well. It'd be used to access `LoggerEnv` and to access the underlying one you'd use `lift ask`.